exposed-migration/src/main/kotlin/MigrationUtils.kt

```diff
@@ -90,6 +91,48 @@ object MigrationUtils {
         return allStatements
     }
 
+    /**
+     * Returns the SQL statements that drop any columns that exist in the database but are not defined in [tables].
+     *
+     * By default, a description for each intermediate step, as well as its execution time, is logged at the INFO level.
+     * This can be disabled by setting [withLogs] to `false`.
+     *
+     * **Note:** Some dialects, like SQLite, do not support `ALTER TABLE DROP COLUMN` syntax completely.
+     * Please check the documentation.
+     */
+    fun dropUnmappedColumnsStatements(vararg tables: Table, withLogs: Boolean = true): List<String> {
+        if (tables.isEmpty()) return emptyList()
+
+        val statements = mutableListOf<String>()
+
+        val dbSupportsAlterTableWithDropColumn = TransactionManager.current().db.supportsAlterTableWithDropColumn
+
+        if (dbSupportsAlterTableWithDropColumn) {
+            val existingTablesColumns = logTimeSpent("Extracting table columns", withLogs) {
+                currentDialect.tableColumns(*tables)
+            }
+
+            val tr = TransactionManager.current()
+
+            tables.forEach { table ->
+                val existingColumns = existingTablesColumns[table].orEmpty().toSet()
+                val tableColumns = table.columns.toSet()
+                val mappedColumns = existingColumns.mapNotNull { columnMetadata ->
+                    val mappedCol = tableColumns.find { column -> columnMetadata.name.equals(column.nameUnquoted(), ignoreCase = true) }
+                    if (mappedCol != null) columnMetadata else null
+                }.toSet()
+                val unmappedColumns = existingColumns.subtract(mappedColumns)
+                unmappedColumns.forEach {
+                    statements.add(
+                        "ALTER TABLE ${tr.identity(table)} DROP COLUMN ${tr.db.identifierManager.quoteIdentifierWhenWrongCaseOrNecessary(it.name)}"
+                    )
```

Comment:
From what I'm seeing, this is what using `Column.dropStatement().single()` directly returns, no?

Reply:
Yes